lte/gateway/c/oai/tasks/s1ap/s1ap_mme_handlers.c

```diff
@@ -447,6 +447,19 @@ int s1ap_mme_handle_s1_setup_request(
         S1ap_TimeToWait_v20s);
     increment_counter(
         "s1_setup", 1, 2, "result", "failure", "cause", "invalid_state");
+    // Check if the UE counters for eNB are equal.
+    // If not, the eNB will never switch to INIT state, particularly in
+    // stateless mode.
+    // Exit the process so that health checker can clean-up all Redis
+    // state and restart all stateless services.
+    AssertFatal(
+        enb_association->nb_ue_associated ==
+            enb_association->ue_id_coll.num_elements,
+        "Num UEs associated with eNB (%u) is more than the UEs with valid "
+        "mme_ue_s1ap_id (%zu), the eNB will never recover from this. "
```

Comment:
Maybe a better message is needed. E.g., instead of saying eNB will never be recover from this, it might be better to say "This is a deadlock state potentially caused by misbehaving eNB; restarting MME. In stateless mode, health management service will eventually detect multiple MME restarts due to this deadlock state and force sctpd and hence all services to restart."